src/Database/LSMTree/Internal.hs

```diff
@@ -449,7 +490,12 @@ mkLevelsCache lvls = LevelsCache_ {
     , cachedIndexes   = V.map Run.runIndex rs
     , cachedKOpsFiles = V.map Run.runKOpsFile rs
     }
-  where rs = V.concatMap residentRuns lvls
+  where
+    rs = flip V.concatMap lvls $ \(Level mr rs') ->
+      V.cons (case mr of
+                SingleRun r                   -> r
+                MergingRun (CompletedMerge r) -> r)
+             rs'
```

Comment:
Hmm, maybe instead of providing `closeLevels` above, it would have been nicer to provide `runsInLevels` and the we could use it here and also do `for_ (runsInLevels lvls) $ Run.removeReference (tableHasFS thEnv)` to close them.

But not too important, just a little bit of code duplication.